msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -365,7 +397,8 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
                 {
                     filteredItems = filteredItems.Where(
                         item => environmentAliases.Contains(item.Environment) &&
-                        item.TenantId.Equals(requestParams.Authority.GetTenantId(), StringComparison.OrdinalIgnoreCase));
+                        ((item.TenantId==null && requestParams.Authority.GetTenantId()==null) ||
```

Comment:
>TenantId [](start = 31, length = 8)

I think the code would be clearer if we put an IsADFS { get { return tenantId==null }} property on the token cache items. We're making this assumption everywhere, so it's breaking SRP.